src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -1,58 +1,388 @@
+
+## Graphsync in ChainSync
+
+`ChainSync` is written in terms of `Graphsync`. `ChainSync` adds blockchain and filecoin-specific
+synchronization functionality that is critical for Filecoin security.
+
+### Rate Limiting Graphsync responses (SHOULD)
+
+When running Graphsync, Filecoin nodes must respond to graphsync queries. Filecoin requires nodes
+to provide critical data structures to others, otherwise the network will not function. During
+ChainSync, it is in operators' interests to provide data structures critical to validating,
+following, and participating in the blockchain they are on. However, this has limitations, and
+some level of rate limiting is critical for maintaining security in the presence of attackers
+who might issue large Graphsync requests to cause DOS.
+
+We recommend the following:
+
+- **Set and enforce batch size rate limits.**
+  Force selectors to be shaped like: `LimitedBlockIpldSelector(blockCID, BatchSize)` for a single
+  constant `BatchSize = 1000`.
+  Nodes may push for this equilibrium by only providing `BatchSize` objects in responses,
+  even for pulls much larger than `BatchSize`. This forces subsequent pulls to be run, re-rooted
+  appropriately, and hints at other parties that they should be requesting with that `BatchSize`.
+- **Force all Graphsync queries for blocks to be aligned along cacheable bounderies.**
+  In conjunction with a `BatchSize`, implementations should aim to cache the results of Graphsync
+  queries, so that they may propagate them to others very efficiently. Aligning on certain boundaries
+  (eg specific `ChainEpoch` limits) increases the likelihood many parties in the network will request
+  the same batches of content.
+  Another good cacheable boundary is the entire contents of a `Block` (`BlockHeader`, `Messages`,
+  `Signatures`, etc).
+- **Maintain per-peer rate-limits.**
+  Use bandwidth usage to decide whether to respond and how much on a per-peer basis. Libp2p already
+  tracks bandwidth usage in each connection. This information can be used to impose rate limits in
+  Graphsync and other Filecoin protocols.
+- **Detect and react to  DOS: restrict operation.**
+  The safest implementations will likely detect and react to DOS attacks. Reactions could include:
+  - Smaller `Graphsync.BatchSize` limits
+  - Fewer connections to other peers
+  - Rate limit total Graphsync bandwidth
+  - Assign Graphsync bandwidth based on a peer priority queue
+  - Disconnect from and do not accept connections from unknown peers
+  - Introspect Graphsync requests and filter/deny/rate limit suspicious ones
+
+
+### NaiveChainsync by just Graphsyncing blocks from BlockPubsub (MAY)
+
+It is highly probable that many nodes will find success by running a protocol
+as simple as this:
+
+```go
+func (c *NaiveChainsync) Sync() {
+  // for each of the blocks we see in the gossip channel
+  for block := range c.BlockPubsub.NewBlocks() {
+    c.IpldStore.Add(block)
```

Comment:
is this assuming that the BlockPubsub validator is checking the high level (stateless) validity of the block?